### PR TITLE
Hide hidden files

### LINKED ISSUE #63: Attempting to save file to nonexistent directory in .thought_box breaks application
Trying to save a file to a directory that does not exist in .thought_box (e.g., `nonexistent/.file.txt`) triggers the popup `[Errno 2: No such file or directory : '<directory-name>'`. After closing the app, restarting  triggers the same error because `editor.py` tries to read the nonexistent path  when the `ThoughtBox` instance is created. I don't know how to reset the application state when this happens.

Recommended fixes:

1. In `save_file_as_path`, add a condition verifying the dirname of the user's entered path either exists or is the empty string (for a file saved at the top level in `.thought_box`.
2. Add an error handler to `ThoughtBox`'s `__init__` to display an empty file if `self.application_state.current_path` can't be read.

### PR DESCRIPTION
**Pull request by BoraxTheClean**
- Hide Hidden Files
- Display a prettier directory name rather than `.thought_box`

### CODE REVIEW

File: src/custom_types/scroll_menu.py [hunk 1]
```diff
@@ -97,11 +101,14 @@ def _get_contents(self, directory: str) -> List[Frame]:
...
-                    handler=functools.partial(self._display_content, item, directory),
+                    text=file_name,
+                    handler=functools.partial(
+                        self._display_content, file_name, directory
+                    ),
                 )
             )
-            for item in listdir(directory)
+            for file_name in listdir(directory)
+            if not file_name.startswith(".")
```
Comment (ghost) [inline src/custom_types/scroll_menu.py]: If I'm reading right the user would still be able to create a file that starts with `.` as long as they enter a path starting from a directory inside `.thought_box` that doesn't itself start with `.` Should we prevent this?

**Review by ghost — COMMENTED** [review]
Good work, though per my comment it might be possible for users to evade the dotfile restriction.

File: src/custom_types/scroll_menu.py (hunk 1 above)
Comment (BoraxTheClean) [reply src/custom_types/scroll_menu.py]: I believe this was addressed in 6833501c3cde74adeff3502ce271317870e08e70.

Comment (BoraxTheClean) [reply src/custom_types/scroll_menu.py]: If you're seeing another way around it, please elaborate. Or record a loom of you doing the thing.

Comment (ghost) [reply src/custom_types/scroll_menu.py]: I think the user could evade it by saving to a path like:
```
"foo/.bar.txt"
```
where the path doesn't start with a `.` but the filename does. We could stop it by changing the condition in  line 136 of `menu_bar.py` to
```
not os.path.split(user_entered_path)[1].starts_with(".")
```

**Comment by SharpBit** [comment]
> Good work, though per my comment it might be possible for users to evade the dotfile restriction.

image
It seems to be possible, yes.

File: src/custom_types/scroll_menu.py (hunk 1 above)
Comment (SharpBit) [reply src/custom_types/scroll_menu.py]: > If I'm reading right the user would still be able to create a file that starts with `.` as long as they enter a path starting from a directory inside `.thought_box` that doesn't itself start with `.` Should we prevent this?

This should be fixed now in f8b34c4756b1f341bf453543a6a694bc87b791de

**Review by ghost — APPROVED** [review]
Only setting `application_state.current_path` if the save is successful is a good idea.

**Pull request merged**